async-nats/src/jetstream/consumer/push.rs

```diff
@@ -264,3 +260,260 @@ impl IntoConsumerConfig for &Config {
+            .await?;
+
+        Ok(Ordered {
+            context: self.context.clone(),
+            consumer: self,
+            subscriber: Some(subscriber),
+            subscriber_future: None,
+            stream_sequence: 0,
+        })
+    }
+}
+
+pub struct Ordered<'a> {
+    context: Context,
+    consumer: Consumer<OrderedConfig>,
+    subscriber: Option<Subscriber>,
+    subscriber_future: Option<BoxFuture<'a, Result<Subscriber, Error>>>,
+    stream_sequence: u64,
+}
+
+impl<'a> futures::Stream for Ordered<'a> {
+    type Item = Result<Message, Error>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> Poll<Option<Self::Item>> {
+        loop {
+            if self.subscriber.is_none() {
+                match self.subscriber_future.as_mut() {
+                    None => {
+                        let context = self.context.clone();
+                        let sequence = self.stream_sequence;
+                        let config = self.consumer.config.clone();
+                        let stream_name = self.consumer.info.stream_name.clone();
+                        self.subscriber_future = Some(Box::pin(async move {
+                            recreate_ephemeral_subscriber(context, config, stream_name, sequence)
+                                .await
+                        }));
+                        match self.subscriber_future.as_mut().unwrap().as_mut().poll(cx) {
+                            Poll::Ready(subscriber) => {
+                                self.subscriber_future = None;
+                                self.subscriber = Some(subscriber?);
+                            }
+                            Poll::Pending => {
+                                return Poll::Pending;
+                            }
+                        }
+                    }
+                    Some(subscriber) => match subscriber.as_mut().poll(cx) {
+                        Poll::Ready(subscriber) => {
+                            self.subscriber_future = None;
+                            self.subscriber = Some(subscriber?);
+                        }
+                        Poll::Pending => {
+                            return Poll::Pending;
+                        }
+                    },
+                }
+            }
+            if let Some(subscriber) = self.subscriber.as_mut() {
+                match subscriber.receiver.poll_recv(cx) {
+                    Poll::Ready(maybe_message) => {
+                        match maybe_message {
+                            Some(message) => {
+                                match message.status {
+                                    Some(StatusCode::IDLE_HEARBEAT) => {
+                                        if let Some(headers) = message.headers.as_ref() {
+                                            let sequence: u64 = headers
+                                            .get("nats-last-stream")
```

Comment:
Add canonical name and constant in the header module

```suggestion
                                            .get(header::NATS_LAST_STREAM)
```